core/config/v2/types.go

```diff
@@ -0,0 +1,263 @@
+package v2
+
+import (
+	"net"
+
+	ocrnetworking "github.com/smartcontractkit/libocr/networking"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/core/config"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+//TODO doc
+type Core struct {
+	// General/misc
+	Dev                 *bool
+	ExplorerURL         *models.URL
+	InsecureFastScrypt  *bool
+	ReaperExpiration    *models.Duration
+	Root                *string
+	ShutdownGracePeriod *models.Duration
+
+	Database *Database
+
+	TelemetryIngress *TelemetryIngress
+
+	Log *Log
+
+	WebServer *WebServer
+
+	//TODO feature table?
+	FeatureFeedsManager *bool
+	FeatureUICSAKeys    *bool
+
+	FeatureLogPoller *bool
+
+	JobPipeline *JobPipeline
+
+	FluxMonitor *FluxMonitor
+
+	FeatureOffchainReporting2 *bool
+	OCR2                      *OCR2
+
+	FeatureOffchainReporting *bool
+	OCR                      *OCR
+
+	P2P *P2P
+
+	Keeper *Keeper
+
+	AutoPprof *AutoPprof
+}
+
+type Secrets struct {
+	DatabaseURL       *models.URL
+	ExplorerAccessKey string `toml:",omitempty"`
+	ExplorerSecret    string `toml:",omitempty"`
+	//TODO more?
+}
+
+type Database struct {
+	ListenerMaxReconnectDuration  *models.Duration
+	ListenerMinReconnectInterval  *models.Duration
+	MigrateOnStartup              *bool
+	ORMMaxIdleConns               *int64
+	ORMMaxOpenConns               *int64
+	TriggerFallbackDBPollInterval *models.Duration
+
+	Lock *DatabaseLock
+
+	Backup *DatabaseBackup
+}
+
+type DatabaseLock struct {
+	Mode                  *string
+	AdvisoryCheckInterval *models.Duration
+	AdvisoryID            *int64
+	LeaseDuration         *models.Duration
+	LeaseRefreshInterval  *models.Duration
+}
+
+type DatabaseBackup struct {
+	Dir              *string
+	Frequency        *models.Duration
+	Mode             *config.DatabaseBackupMode
+	OnVersionUpgrade *bool
+	URL              *models.URL
+}
+
+type TelemetryIngress struct {
+	UniConn      *bool
+	Logging      *bool
+	ServerPubKey *string
+	URL          *models.URL
+	BufferSize   *uint16
+	MaxBatchSize *uint16
+	SendInterval *models.Duration
+	SendTimeout  *models.Duration
+	UseBatchSend *bool
+}
+
+type Log struct {
+	JSONConsole    *bool
+	FileDir        *string
+	Level          *zapcore.Level //TODO is this actually an exceptional case to leave as env var?
+	SQL            *bool
+	FileMaxSize    *utils.FileSize
+	FileMaxAgeDays *int64
+	FileMaxBackups *int64
+	UnixTS         *bool
```

Comment:
Same for `LOG_SQL_MIGRATIONS` w/ goose.

Reply:
Possible to refactor to make these use the config?

Reply:
Definitely not without changing things, and would potentially require an extreme refactor and create a dependency nightmare, since low level package init funcs depend on the config that depends on everything else. Alternatively, we could explore moving those out of init funcs, but I'm not certain we can actually avoid races, even if we control the initial go routine.

Reply:
Yeah lets not bother then. Leave as env vars.